File: _c++/symulacja_organizmow/Organizmy/Zwierzeta/Antylopa.cpp
//
//  Antylopa.cpp
//  WirtualnySwiat
//
//  Created by Michal Tarnacki on 10/04/2022.
//

#include "Antylopa.hpp"

Antylopa::Antylopa(Swiat &swiat):Zwierze(swiat){
    SetInicjatywa();
    SetSila();
};
void Antylopa::Akcja(){
    Ruch(2);
}
void Antylopa::Kolizja(Organizm& organizm){
    bool czySieRuszyl;
    int x=rand()%100;
    if(x>=50&&!dynamic_cast<Antylopa*>(&organizm)){
        czySieRuszyl=RuchBezKolizji(2);
        if(!czySieRuszyl)
            Zwierze::Kolizja(organizm);
        else{
            std::cout << "Antylopa: " << this->GetPolozenie() << " uciekae bez kolizji" << std::endl;
        }
    }
    else
        Zwierze::Kolizja(organizm);
}
char Antylopa::Rysowanie() const{
    return 'A';
}
void Antylopa::SetInicjatywa(){
    this->inicjatywa=4;
}
void Antylopa::SetSila(){
    this->sila=4;

}
Organizm* Antylopa::Urodz(){
    return new Antylopa(this->swiat);
}
bool Antylopa::MozeBycPartnerem(Organizm& organizm){
    if(dynamic_cast<Antylopa*>(&organizm))
       return true;
    else
       return false;
}

File: _java/symulacja_organizmow/src/Symulacja/Zwierzeta/Antylopa.java
package Symulacja.Zwierzeta;
import Symulacja.*;

import java.awt.*;

public class Antylopa extends Zwierze {
    protected void SetInicjatywa(){
        this.inicjatywa=4;
    };
    protected void SetSila(){
        this.sila=4;
    };
    public Antylopa(Swiat swiat){
        super(swiat);
        SetInicjatywa();
        SetSila();
    };
    public char Rysowanie(){
        return 'A';
    };
    public Organizm Urodz() {
        return new Antylopa(this.swiat);
    };
    public boolean MozeBycPartnerem(Organizm organizm){
        if(organizm instanceof Antylopa)
            return true;
        else
            return false;
    };
    public void Akcja(){
        Ruch(2);
    };
    public void Kolizja(Organizm organizm){
        boolean czySieRuszyl;
        int x=(int)(Math.random()*100);
        if(x>=50&&!(organizm instanceof Antylopa)){
            czySieRuszyl=RuchBezKolizji(2);
            if(!czySieRuszyl)
                super.Kolizja(organizm);
        else{
                swiat.zdarzenia.DodajString( "Antylopa: " + this.GetPolozenie() + " ucieka bez kolizji");
            }
        }
    else
        super.Kolizja(organizm);
    };
    @Override
    public Color GetColor(){
        return new Color(0,255,255);
    };
}
